### ISSUE TITLE
Error generating "ManyThrough" with explicit FK in through-models "BelongsTo"

### ISSUE THREAD

**Issue by proreact-mj** [strongloop/loopback-sdk-angular#187]
Soo. 
I'm trying to generate my Angular SDK, but keep geeting these for my M:N (ManyThrough) relations
_Warning: scope [from model].[to model] targets class "["belongs" relation name]", which is not exposed via remoting. The Angular code for this scope won't be generated._

Note that this is during testing and initial setup, all models, even the relation models are public and everything works from the explorer. It is just the Angular SDK acting up. I'm also a bit puzzed about the target class being stated as the through model's relation name, and not the relation name of the source model. 

After some testing, I found that if I remove the explicit FK from the through models "belongs" relation to work, but becomes a hard constraint, as the relation needs to be named exactly after the property that is FK (guessing the keyfield to be relation name +"Id").

All out removing the "belongs" relation in the through model still seems to work and while I can live with that for now, I see this becoming a problem further along the way.

Is this a bug/shortcomming in the SDK generation or am I missing something?

**Comment by proreact-mj** [comment]
https://github.com/proreact-mj/loopback-sandbox

Further testing:
I manually created models (phys, client, appointment) with slc, for this it works. 
I added the models from my own project (Resident, Activity and ActivityResponsibleResident) these have been migrated from a MySQL, but not attached to the default "db" source. 
phys has many clients through appointment and reverse
Resident has many Activity through ActivityResponsibleResident and reverse
appointment/ActivityResponsibleResident  belongs to each of the models that go through it.

lb-ng server/server.js lbs.js says
Warning: scope Activity.ResponsibleResidents targets class "Residents", which is not exposed
via remoting. The Angular code for this scope won't be generated.
Warning: scope Resident.ResponsibleActivities targets class "Activities", which is not exposed
via remoting. The Angular code for this scope won't be generated.

The explorer/swagger shows Activity.ResponsibleResidents and Resident.ResponsibleActivities fine.
Removing the "belongs to" on ResponsibleActivities makes the lb-ng work, but then the manyThrough relation only work on the condition that the FK in ResponsibleActivities is named as "[model]Id".

And again underlined that this is not a problem for an ad hoc created model, only in the models that I migrated from a MySQL DB

**Comment by gustavomick** [comment]
trying to add more info, i have this example:

ent.json 

```
{
  "name": "ent",
  "base": "PersistedModel",
  "idInjection": true,
  "options": {
    "validateUpsert": true
  },
  "properties": {
    "name": {
      "type": "string",
      "required": true
    },
  },
  "validations": [],
  "relations": {
    "children": {
      "type": "hasMany",
      "model": "ent",
      "keyThrough": "entId",
      "foreignKey": "parentId",
      "through": "entGraph"
    },
    "parents": {
      "type": "hasMany",
      "model": "ent",
      "keyThrough": "parentId",
      "foreignKey": "entId",
      "through": "entGraph"
    }
  },
  "acls": [],
  "methods": {}
}
```
ent-graph.json

```
{
  "name": "entGraph",
  "base": "PersistedModel",
  "idInjection": true, 
  "options": {  
    "validateUpsert": true 
  }, 
  "properties": {},
  "validations": [],
  "relations": {
    "ent": {
      "type": "belongsTo",
      "model": "ent" 
    },    
    "parent": {
      "type": "belongsTo",
      "model": "ent"
    }
  },
  "acls": [],
  "methods": {}
}
```
i made a "self referenced" model (like a graph) both working at api/explorer level ({"include":["children","parents"]}) but at lb-ng level Ent.children is generated but Ent.parents is not. 

ref: https://docs.strongloop.com/display/public/LB/HasManyThrough+relations#HasManyThroughrelations-Selfthrough

**Comment by proreact-mj** [comment]
Any chance this bug will get some love soon? It is a bit of a hassel for me to maintain a branch with "wrong code" just to enable lb-ng generator functionality.

**Comment by Undrium** [comment]
Yeah, this problem is causing alot of trouble for us aswell. We are using the "Loopback Component Group Access"-module to enable group access to our resources, this module requires us to use "through"-relations which aren't properly generated in the SDK-file. It's painful to program around (we have to do manual API-calls mixed with the sdk-functions) something so fundamental.

**Comment by proreact-mj** [comment]
I agree that both issues addresses the same core issue, but is it right to close it if it isn't fixed or none-issue and you are welcoming contributions? (no critique, just trying to understand the process)

**Comment by bajtos** [comment]
@proreact-mj we are trying to use GitHub issue tracker as a backlog of things we would like to work on in the near future (let's say 3-6 months). Since this issue did not get enough priority for the last 12+ months, I find it very unlikely that we will ever get to work on it. By keeping it open, we would have to ignore it/filter it out whenever we look at our backlog, which is rather cumbersome.

**Comment by proreact-mj** [comment]
Thanks for the explanation/clarification. 
My only grief is that, closing it will increase the likelihood of new people thinking this has been fixed, opening a new issues with the same problem or marking it as regression, as well as making it near impossible for contributes to identify it when looking for stuff to do.
